### DISCUSSION TITLE
How can I close a Popover after it opens some other content?

### DISCUSSION THREAD

**Discussion by GustavoSmith** [adobe/react-spectrum#5717] [Q&A]
[screen-capture.webm](https://github.com/adobe/react-spectrum/assets/51140828/5b22e042-62ea-45d6-a7cd-4219844b6a67)

I'm using Tailwind and React Aria Components. I wanted to display a Modal triggered by a Button inside a Popover, but when I open the Modal, the Popover doesn't close (and in this case, it can overlap). Using "aria-hidden:hidden" on the Popover solves this, but I was wondering if there's a better approach. I've tried controlling the state of the Modal and the Popover, changing first the state of one and then the other, but couldn't make it work.

This is my code:

```
<DialogTrigger>
      <Button className="bg-green-300 rounded-full p-4">First Button</Button>
      <Popover placement="bottom end" className="fixed inset-0 z-10 flex min-h-full items-center justify-center overflow-y-auto">
        <Dialog className="w-[clamp(21.25rem,90vw,34.25rem)] rounded h-32 flex flex-col justify-center items-center bg-red-400 shadow-xl focus:outline-none">
          <p>Some content here...</p>
          <DialogTrigger>
            <Button className="rounded text-green-200 bg-blue-700 p-4 bg-ecomm-blue-100 outline-none hover:bg-ecomm-blue-200">
              A button that opens something else
            </Button>
            <ModalOverlay className="fixed inset-0 z-10 flex min-h-full items-center justify-center overflow-y-auto bg-black/25 p-4 text-center backdrop-blur">
              <Modal className="w-full max-w-md overflow-hidden rounded-2xl bg-white p-6 text-left align-middle shadow-xl">
                <Dialog className="relative outline-none">
                  {({ close }) => (
                    <>
                      <Heading
                        slot="title"
                        className="text-xxl my-0 font-semibold leading-6 text-slate-700"
                      >
                        Do the important action.
                      </Heading>

                      <p className="mt-3 text-slate-500">
                        Are you sure you want to do this?
                      </p>
                      <div className="mt-6 flex justify-end gap-2">
                        <Button className="" onPress={close}>
                          Of course.
                        </Button>
                      </div>
                    </>
                  )}
                </Dialog>
              </Modal>
            </ModalOverlay>
          </DialogTrigger>
        </Dialog>
      </Popover>
    </DialogTrigger>
```

I couldn't provide a Codesandbox showing this because I couldn't make Tailwind work there, if it helps please let me know. Thank you!

**Comment by reidbarber** [comment]
If you want to display the inside one on top of the outside one, I think you can just up the z-index of the inside one (i.e. `z-10` -> `z-20`). If you want to completely close the outside one before opening the inside one, I think you will need to have a completely different setup where the modal is controlled and is outside the Popover and DialogTrigger.

**Reply by GustavoSmith** [reply]
Thank you so much! Changing this setup makes sense semantically though? Since the modal it's related to the popover, it's straight-foward to put one inside of the other. That was my only concern, since both your approach and mine "work". Would an arbitrary "aria" value help with this? Or am I mixing concepts?
